### PR TITLE
Enable ASP.NET Core installers on Linux in the VMR

### LINKED ISSUE #4692: aspnetcore isn't building installers in the VMR
The `BuildInstallers` switch defines if installers should be built which defaults to false. We should pass the switch in here: https://github.com/dotnet/dotnet/blob/eb82b31516b032924126ac2c81f3edca80911619/repo-projects/aspnetcore.proj#L12

-buildinstallers (Unix) / - BuildInstallers (Windows)

Note that there's also a join point on Windows in the installers which will fail the Windows dev leg when it gets enabled: https://github.com/dotnet/aspnetcore/issues/52884

cc @jkoritzinsky @NikolaMilosavljevic

### PR DESCRIPTION
**Pull request by jkoritzinsky**
The new tooling has flowed, so we can enable this now.

Fixes dotnet/source-build#4692

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]
Wonderful 👍

File: src/SourceBuild/content/repo-projects/aspnetcore.proj [hunk 1]
```diff
@@ -9,9 +9,11 @@
...
     <BuildActions Condition="'$(BuildOS)' != 'windows'">$(BuildActions) $(FlagParameterPrefix)no-build-java</BuildActions>
-    <!-- Enable building installers on Windows. We can't build installers on Linux until https://github.com/dotnet/aspnetcore/pull/58612 as been merged and flows -->
+    <!-- Enable building installers on Windows and Linux. ASP.NET Core doesn't produce installers on Mac -->
     <BuildActions Condition="'$(BuildOS)' == 'windows'">$(BuildActions) $(FlagParameterPrefix)BuildInstallers</BuildActions>
-    <!-- <BuildActions Condition="'$(BuildOS)' != 'windows'">$(BuildActions) $(FlagParameterPrefix)build-installers</BuildActions> -->
+    <BuildActions Condition="'$(BuildOS)' == 'linux'">$(BuildActions) $(FlagParameterPrefix)build-installers</BuildActions>
+    <!-- In a source-only build, we don't pass -all, so we need to explicitly opt-in to managed components here. -->
+    <BuildActions Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(BuildOS)' == 'linux'">$(FlagParameterPrefix)build-managed</BuildActions>
```
Comment (ViktorHofer) [inline src/SourceBuild/content/repo-projects/aspnetcore.proj:16]: Interesting. How did this work before?

Comment (jkoritzinsky) [reply src/SourceBuild/content/repo-projects/aspnetcore.proj:16]: When none of the `--buildX` arguments or `-all` are passed, `--build-managed` is the default. So by adding `--build-installers`, we effectively removed the implicit `--build-managed`. That's why it was only failing in source-build.

Comment (ViktorHofer) [reply src/SourceBuild/content/repo-projects/aspnetcore.proj:16]: Ah yes, makes sense. Thanks.

**Review by mmitche — APPROVED** [review]

**Pull request merged**